src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -272,11 +209,112 @@ internal void StartPushingToWorkspaceAndNotifyOfOpenDocuments(IEnumerable<Abstra
         {
```

Comment:
I guess projects given is already in right order to push?

Reply:
Note this code is largely moving from VisualStudioWorkspaceTracker.WorkspaceHostState. If there's a better name I can use happy to fix it.